### PR TITLE
Add vertx4 mysql client instrumentation

### LINKED ISSUE #5716: Vertx 4 MySQL client instrumentation
The current vertx MySQL instrumentation does not support vertx4.

### PR DESCRIPTION
**Pull request by akshaypatidar1999**
# What Does This Do
- Add instrumentation for vertx MySQL client for vertx version `4.0.0-4.1.0` and `4.4.2-4.4.4`. Some instrumentation needs to be added for vertx version `4.1.0-4.4.2`

# Additional Notes
- Implements #5716

### CODE REVIEW

**Comment by akshaypatidar1999** [comment]
Not sure whether following muzzle confiuration can be removed. Keeping this causes the tests to fail.
```
fail {
    group = 'io.vertx'
    module = 'vertx-pg-client'
    versions = '[,]'
}
```

**Comment by akshaypatidar1999** [comment]
Upon testing this with vertx 4.4.4 I noticed that the spans do not contain full information like `db.user`, `db.type` etc. On digging in the code I found that the `MySQLPoolImpl` constructor signature has changed. I think we will have to create a separate module for vertx4 MySQL client. Will update the PR with the same.

**Comment by akshaypatidar1999** [comment]
While trying to implement support for vertx 4 MySQL client I noticed that the signature of implementations of MySQLPool has changed in vertx version 4.0.0, 4.1.0, 4.2.0 and 4.4.0 so it would not be possible to have an implementation that will work with all vertx versions.

Currently, we are using vertx 4.4.4, so is it okay if I just add an implementation for vertx versions [4.4.0, 4.5.0)?

**Comment by akshaypatidar1999** [comment]
Not sure why the tests are failing. Can someone please help me identify the issue?

**Comment by bm1549** [comment]
Hi @akshaypatidar1999 👋 thank you for contributing

It looks like some of the [tests you introduced](dd-java-agent/instrumentation/vertx-mysql-client-4.0/src/test/groovy/VertxSqlClientForkedTest.groovy) are [failing](https://app.circleci.com/pipelines/github/DataDog/dd-trace-java/28254/workflows/b60331a8-a937-47e6-a6bc-85a2fe052349/jobs/942620/tests). It looks like they're failing due to various reasons such as mismatched operation names and trace counts

If you run those tests locally, you should be able to reproduce the test failure

**Comment by akshaypatidar1999** [comment]
> 

@bm1549 How do I run tests in local? I am running the following commands as mentioned [here](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#verifying-instrumentations). This command executes successfully. I guess the tests are not being executed with this command.
```
./gradlew :dd-java-agent:instrumentation:vertx-mysql-client-4.0:test
./gradlew :dd-java-agent:instrumentation:vertx-mysql-client-4.4.2:test
```

**Comment by PerfectSlayer** [comment]
And don’t forget to install the [pre-commit hook](https://github.com/DataDog/dd-trace-java/blob/master/CONTRIBUTING.md#pre-commit-hook) if you want to avoid discovering formatting error on CI jobs 😉

**Comment by akshaypatidar1999** [comment]
@bm1549 I have added tests for  `vertx-mysql-client-4.0` and they are executing successfully. The module `vertx-mysql-client-4.4.2` contains instrumentation that will be applied on top of instrumentation in `vertx-mysql-client-4.0`. As this does not contain all the instrumentation needed how do I write tests for this?

**Comment by akshaypatidar1999** [comment]
@bm1549 @PerfectSlayer Can you please review this PR?

**Comment by akshaypatidar1999** [comment]
@bm1549 The `Lib Injection Test` [workflow](https://github.com/DataDog/dd-trace-java/actions/runs/5947928997/job/16214313404) that you ran is failing due to some docker registry access issue. Can you please check?

File: dd-java-agent/instrumentation/vertx-mysql-client-4.0/src/main/java/datadog/trace/instrumentation/vertx_sql_client_4/SqlClientBaseInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package datadog.trace.instrumentation.vertx_sql_client_4;
+
+import static datadog.trace.agent.tooling.bytebuddy.matcher.NameMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.*;
```
Comment (nayeem-kamal) [inline dd-java-agent/instrumentation/vertx-mysql-client-4.0/src/main/java/datadog/trace/instrumentation/vertx_sql_client_4/SqlClientBaseInstrumentation.java:4]: Same as before, explicit imports would be best here.

**Comment by akshaypatidar1999** [comment]
> Overall looks great! Thanks for your contribution. I just marked some small changes to be made before merging.

Thanks for the feedback @nayeem-kamal. I have updated the PR with the suggestions

**Comment by akshaypatidar1999** [comment]
@bm1549 @nayeem-kamal The Lib injection [workflow](https://github.com/DataDog/dd-trace-java/actions/runs/6150421886/job/16734853205?pr=5717) failed again due to some docker registry access issue. However, the same workflow works for other [PR](https://github.com/DataDog/dd-trace-java/pull/5880). Let me know if something needs to be changed in this PR.

**Review by nayeem-kamal — APPROVED** [review]
LGTM!

**Pull request merged**